docs/architecture/dapr-for-net-developers/secrects-buildingblock.md

```diff
@@ -0,0 +1,553 @@
+string connectionString = secrets["customerdb"];
+```
+
+Arguments for the `GetSecretAsync` method include:
+
+- The name of the Dapr secret store component ('secrets-store')
+- The secret to retrieve ('eshopsecrets')
+- Optional metadata key/value pairs ('version_id=3')
+
+The method responds with a dictionary object as a secret can contain multiple key/value pairs. In the example above, the secret named `customerdb` is referenced from the collection to return a connection string.
+
+The Dapr .NET SDK also features a .NET configuration provider. It loads specified secrets into the underlying [.NET Core configuration API](https://docs.microsoft.com/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0). The running application can then reference secrets from the `IConfiguration` dictionary that is registered in ASP.NET Core dependency injection.
+
+The secrets configuration provider is available from the [Dapr.Extensions.Configuration](https://www.nuget.org/packages/Dapr.Extensions.Configuration) NuGet package. The provider can be registered in the `Program.cs` of an ASP.NET Web API application:  
+
+```csharp
+public static IHostBuilder CreateHostBuilder(string[] args) =>
+    Host.CreateDefaultBuilder(args)
+        .ConfigureAppConfiguration(config =>
+        {
+            var daprClient = new DaprClientBuilder().Build();
+            var secretDescriptors = new List<DaprSecretDescriptor>
+            {
+                new DaprSecretDescriptor("eshopsecrets")
+            };
+            config.AddDaprSecretStore("secrets-store", secretDescriptors, daprClient);
+        })
+        .ConfigureWebHostDefaults(webBuilder =>
+        {
+            webBuilder.UseStartup<Startup>();
+        });
+```
+
+The above example loads the `eshopsecrets` secrets collection into the .NET configuration system at startup. Registering the provider requires an instance of `DaprClient` to invoke the secrets API on the Dapr sidecar. The other arguments include the name of the secrets management component and a `DaprSecretDescriptor` object with the name of the secret.
+
+Once loaded, you can retrieve secrets directly from application code:
+
+```csharp
+public void GetCustomer(IConfiguration config)
+{
+    var connectionString = config["eshopsecrets"]["customerdb"];
+}
+```
+
+## Secrets management components
+
+The secrets management building block supports several secret store components. At the time of writing, the following secret stores are available:
+
+- Environment Variables
+- Local file
+- Kubernetes secrets
+- AWS Secrets Manager
+- Azure Key Vault
+- GCP Secret Manager
+- HashiCorp Vault
+
+ > [!IMPORTANT]
+ > The environment variables and local file components are designed for development workloads only.
+
+The following sections show how to configure a secrets management component.
+
+### Configuration (TODO Move generic text to getting started)
```

Comment:
```suggestion
### Configuration
```

Not sure if this TODO was done or not?

Reply:
We will do this during a last pass over the entire book.